### ISSUE TITLE
[1.3-rc.1] Can't get phantomjs-prebuilt path when a published package depends on it

### ISSUE THREAD

**Issue by aldeed** [meteor/meteor#6496]
### Background
To enable running client side tests in phantomjs with Meteor 1.3, I published a driver package (https://github.com/DispatchMe/meteor-mocha-phantomjs) which depends on https://github.com/DispatchMe/meteor-phantomjs-tests to install and run phantomjs. phantomjs-tests depends on the NPM package `phantomjs-prebuilt`.

So I do `npm i --save-dev mocha`, then `meteor add dispatch:mocha-phantomjs`.

Then to run tests: `meteor test --once --driver-package dispatch:mocha-phantomjs`

### Problem

The phantomjs path is provided by `phantomjs-prebuilt` in `phantomjs.path` (see https://github.com/DispatchMe/meteor-phantomjs-tests/blob/master/server.js#L20).

When the `dispatch:phantomjs-tests` is copied locally into `/packages`, the path is `/Users/eric/DevProjects/dispatch/meteor-notifications-workers/packages/dispatch-phantomjs-tests/.npm/package/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs` and everything works fine.

When the published `dispatch:phantomjs-tests` is used, the path is `/node_modules/meteor/dispatch:phantomjs-tests/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs` and this of course does not exist.

@benjamn has explained about the `/node_modules` fake path before, so this is not too surprising, but it is a little surprising that it works fine with local packages.

### Solutions

Since there is a workaround, this isn't critical to fix, but in general there must be a way to fix this or to give packages a function for translating the fake `/node_modules` path into the real path.

cc @tmeasday

**Comment by tmeasday** [comment]
The issue is on thisline: https://github.com/Medium/phantomjs/blob/master/lib/phantomjs.js#L18, 

```js
 __dirname === "/node_modules/meteor/dispatch:phantomjs-tests/node_modules/phantomjs-prebuilt/lib"
```

Whereas, (if you run `meteor test --test-app-path=/tmp/test-app` at least) it should be

```js
 __dirname === "/tmp/test/app/.meteor/local/build/programs/server/npm/node_modules/meteor/dispatch_phantomjs-tests/npm/node_modules/phantomjs-prebuilt/lib"
```

(noting that `/tmp/test/app/.meteor/local/build/programs/server/npm/node_modules/meteor/dispatch_phantomjs-tests/npm/node_modules -> `~/.meteor/packages/dispatch_phantomjs-tests/.0.0.2.1trwnuh++os+web.browser+web.cordova/npm/npm/node_modules`)

**Comment by tmeasday** [comment]
Ok, reproduced even more simply with a non-test package: https://github.com/tmeasday/dirname-test-app (using https://github.com/tmeasday/meteor-dirname-test | https://atmospherejs.com/tmeasday/dirname-test).

When I run that app I see:

```
`/node_modules/meteor/tmeasday:dirname-test/node_modules/dirname-test
```

**Comment by tmeasday** [comment]
@benjamn I stepped through and found that there's an error being thrown on this line:

https://github.com/meteor/meteor/blob/release-1.3/packages/modules-runtime/modules-runtime.js#L63

It comes from NPM, specifically `Module._resolveFilename:340`. In that function (not sure where to find the online source for it),

```js
request = "./npm/node_modules/meteor/tmeasday_dirname-test/node_modules/dirname-test/package.json"
parent = {
filename: "/private/tmp/dirname-test-app/.meteor/local/build/programs/server/npm-require.js",
id:  "/private/tmp/dirname-test-app/.meteor/local/build/programs/server/npm-require.js"
}
```

You can repro and follow it through yourself here: https://github.com/tmeasday/dirname-test-app

**Comment by tmeasday** [comment]
Could this also be the source of https://github.com/meteor/meteor/issues/6449 also @benjamn?

**Comment by benjamn** [comment]
This was fixed by https://github.com/meteor/meteor/commit/ff023e9c3777c907c4591f9a8485ca9aa7bd45bd, which is the same commit that fixed #6449. Available now in `1.3-rc.3`!
